src/main/kotlin/earth/terrarium/techarium/common/blocks/entities/ComponentBlockEntity.kt

```diff
@@ -0,0 +1,50 @@
+package earth.terrarium.techarium.common.blocks.entities
+
+import net.minecraft.core.BlockPos
+import net.minecraft.core.component.*
+import net.minecraft.world.level.block.entity.BlockEntity
+import net.minecraft.world.level.block.entity.BlockEntityType
+import net.minecraft.world.level.block.state.BlockState
+import net.neoforged.neoforge.common.MutableDataComponentHolder
+
+abstract class ComponentBlockEntity(
+    type: BlockEntityType<*>,
+    pos: BlockPos,
+    state: BlockState
+) : BlockEntity(type, pos, state), MutableDataComponentHolder {
+
+    private val initialComponents: DataComponentMap = DataComponentMap.builder().apply(::createDefaultComponents).build()
+
+    abstract fun createDefaultComponents(builder: DataComponentMap.Builder)
+
```

Comment:
this can be either but as an abstract function this is more inline with the block state registration. Turning this into a abstract property will just make it more annoying on the implementors end

Reply:
`T` implicitly `: Any?`

Reply:
oh wait wrong line